src/silo/prepare_dataset.cpp

```diff
@@ -115,10 +117,26 @@ std::unordered_map<std::string, silo::preprocessing::PartitionChunk> partitionMe
       getMetadataWritersForChunks(metadata_partition_filenames, metadata_reader.reader);
 
    return writeMetadataChunks(
-      alias_key, pango_to_chunk, metadata_reader.reader, chunk_to_metadata_writers, database_config
+      alias_key,
+      pango_to_chunk,
+      metadata_reader.reader,
+      chunk_to_metadata_writers,
+      primary_key_field,
+      partition_by_field
    );
 }
 
+void copyMetadataFile(
+   csv::CSVReader& metadata_reader,
+   silo::preprocessing::MetadataWriter& metadata_writer
+) {
+   metadata_writer.writeHeader(metadata_reader);
+   for (auto& row : metadata_reader) {
+      std::this_thread::sleep_for(std::chrono::nanoseconds(2));
+      metadata_writer.writeRow(row);
+   }
+}
```

Comment:
Can't we simply copy the file? Without reading it line by line using a CSV reader?